File: Baekjoon/BaaarkingDog/0x03_array/5397.py
# 5397: 키로거

N = int(input())

for _ in range(N):
  input_log = input()

  st_left = []
  st_right = []

  for key in input_log:
    if (key == '<'):
      if (len(st_left) > 0): st_right.append(st_left.pop())
    elif (key == '>'):
      if (len(st_right) > 0): st_left.append(st_right.pop())
    elif (key == '-'):
      if (len(st_left) != 0): st_left.pop()
    else: st_left.append(key)
  
  while (len(st_left) > 0): st_right.append(st_left.pop())
  while (len(st_right) > 0): print(st_right.pop(), end="")
  print()

File: Baekjoon/BaaarkingDog/0x03_array/5397.cpp
// 5397: 키로거

#include <iostream>
#include <string>
#include <stack>
using namespace std;

int main() {
  int N;
  cin >> N;

  while (N--) {
    string input_log;
    cin >> input_log;

    stack<char> st_left;
    stack<char> st_right;
    
    for (char key : input_log) {
      if (key == '<') {
        if (st_left.size() > 0) {
          st_right.push(st_left.top());
          st_left.pop();
        }
      } else if (key == '>') {
        if (st_right.size() > 0) {
          st_left.push(st_right.top());
          st_right.pop();
        }
      } else if (key == '-') {
        if (!st_left.empty()) st_left.pop();
      } else {
        st_left.push(key);
      }
    }

    while (st_left.size()) { 
      st_right.push(st_left.top());
      st_left.pop();
    }

    while (!st_right.empty()) {
      cout << st_right.top();
      st_right.pop();
    }

    cout << '\n';
  }


  return 0;
}